hs_tools_resource/app_launch_helper.py

```diff
@@ -13,8 +13,8 @@ def resource_level_tool_urls(resource_obj, request_obj):
     resource_level_app_counter = 0
 
     # associate resources with app tools using extended metadata name-value pair with 'appkey' key
-    filterd_res_obj = BaseResource.objects.filter(short_id=resource_obj.short_id,
-                                                  extra_metadata__has_key=tool_app_key).first()
+    filterd_res_obj = BaseResource.objects.exclude(resource_type='ToolResource').filter(
```

Comment:
Couple of things come to mind:

- we can use get instead of filter and first.
- The app key logic is bypassing the resource type constraints.  This seems like a bug.  If the appkey logic did not bypass the resource type constraint, the web app would not show open with.